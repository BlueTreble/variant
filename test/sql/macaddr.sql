\set ECHO 0
\set QUIET 1
/*
 * NOTE! Unlike the other cases of duplication, we need to handle numeric,
 * float and real separately because of odd things such as
 *
 * 0.0::float < 0.0::real = true
 */

\set baseline_type macaddr
-- Space separated string
\set test_types 'macaddr'
-- Used in array[ :base_values ]
\set base_values '$$08:00:2b:01:02:03$$, $$08:10:2b:01:02:03$$, $$08:20:2b:01:02:03$$, NULL'
-- TODO: Allow more compare values so we can verify we're not somehow screwing up bit ordering
\set compare_value 08:10:2b:01:02:03


\i test/helpers/type_setup.sql

SELECT plan(sum(test_count)::int) FROM plan;

\i test/helpers/type.sql


-- vi: expandtab sw=2 ts=2
/*
 * Box doesn't have sort operators, so do text-based sorting for the actual values.
 *
 * Note that this is used as a string value!
 */
\set op_test_order_by 'ORDER BY compare_type, base_type, compare_value::text, base_value::text'

-- Box also doesn't have a != operator
DELETE FROM operator WHERE op ~ '!=';
