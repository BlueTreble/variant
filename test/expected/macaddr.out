\set ECHO none


ok 1..0


1..90
ok 1 - No NULL variants
ok 2 - Verify result of cast '08:00:2b:01:02:03'::macaddr
ok 3 - Verify result of cast '08:10:2b:01:02:03'::macaddr
ok 4 - Verify result of cast '08:20:2b:01:02:03'::macaddr
ok 5 - Verify result of cast NULL::macaddr
ok 6 - SET base_variant_b
ok 7 - Verify text_in(text_out()) works for (macaddr,08:00:2b:01:02:03)
ok 8 - Verify text_in(text_out()) works for (macaddr,08:10:2b:01:02:03)
ok 9 - Verify text_in(text_out()) works for (macaddr,08:20:2b:01:02:03)
ok 10 - Verify text_in(text_out()) works for (macaddr,)
ok 11
ok 12
ok 13
ok 14
ok 15 - No NULL variants
ok 16 - Verify result of cast '08:10:2b:01:02:03'::macaddr
ok 17 - SET compare_variant_b
ok 18 - Verify text_in(text_out()) works for (macaddr,08:10:2b:01:02:03)
ok 19 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <   '08:00:2b:01:02:03'::macaddr
ok 20 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <   '08:10:2b:01:02:03'::macaddr
ok 21 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <   '08:20:2b:01:02:03'::macaddr
ok 22 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <   NULL::macaddr
ok 23 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <=  '08:00:2b:01:02:03'::macaddr
ok 24 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <=  '08:10:2b:01:02:03'::macaddr
ok 25 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <=  '08:20:2b:01:02:03'::macaddr
ok 26 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr <=  NULL::macaddr
ok 27 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  =  '08:00:2b:01:02:03'::macaddr
ok 28 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  =  '08:10:2b:01:02:03'::macaddr
ok 29 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  =  '08:20:2b:01:02:03'::macaddr
ok 30 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  =  NULL::macaddr
ok 31 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  >= '08:00:2b:01:02:03'::macaddr
ok 32 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  >= '08:10:2b:01:02:03'::macaddr
ok 33 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  >= '08:20:2b:01:02:03'::macaddr
ok 34 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr  >= NULL::macaddr
ok 35 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr   > '08:00:2b:01:02:03'::macaddr
ok 36 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr   > '08:10:2b:01:02:03'::macaddr
ok 37 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr   > '08:20:2b:01:02:03'::macaddr
ok 38 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr   > NULL::macaddr
ok 39 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr !=  '08:00:2b:01:02:03'::macaddr
ok 40 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr !=  '08:10:2b:01:02:03'::macaddr
ok 41 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr !=  '08:20:2b:01:02:03'::macaddr
ok 42 - check casted vs uncasted for '08:10:2b:01:02:03'::macaddr !=  NULL::macaddr
ok 43 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <   '08:00:2b:01:02:03'::macaddr
ok 44 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <   '08:10:2b:01:02:03'::macaddr
ok 45 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <   '08:20:2b:01:02:03'::macaddr
ok 46 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <   NULL::macaddr
ok 47 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <=  '08:00:2b:01:02:03'::macaddr
ok 48 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <=  '08:10:2b:01:02:03'::macaddr
ok 49 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <=  '08:20:2b:01:02:03'::macaddr
ok 50 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr <=  NULL::macaddr
ok 51 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  =  '08:00:2b:01:02:03'::macaddr
ok 52 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  =  '08:10:2b:01:02:03'::macaddr
ok 53 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  =  '08:20:2b:01:02:03'::macaddr
ok 54 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  =  NULL::macaddr
ok 55 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  >= '08:00:2b:01:02:03'::macaddr
ok 56 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  >= '08:10:2b:01:02:03'::macaddr
ok 57 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  >= '08:20:2b:01:02:03'::macaddr
ok 58 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr  >= NULL::macaddr
ok 59 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr   > '08:00:2b:01:02:03'::macaddr
ok 60 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr   > '08:10:2b:01:02:03'::macaddr
ok 61 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr   > '08:20:2b:01:02:03'::macaddr
ok 62 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr   > NULL::macaddr
ok 63 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr !=  '08:00:2b:01:02:03'::macaddr
ok 64 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr !=  '08:10:2b:01:02:03'::macaddr
ok 65 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr !=  '08:20:2b:01:02:03'::macaddr
ok 66 - check variant vs uncasted for '08:10:2b:01:02:03'::macaddr !=  NULL::macaddr
ok 67 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <   '''08:00:2b:01:02:03''::macaddr'
ok 68 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <   '''08:10:2b:01:02:03''::macaddr'
ok 69 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <   '''08:20:2b:01:02:03''::macaddr'
ok 70 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <   'NULL::macaddr'
ok 71 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <=  '''08:00:2b:01:02:03''::macaddr'
ok 72 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <=  '''08:10:2b:01:02:03''::macaddr'
ok 73 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <=  '''08:20:2b:01:02:03''::macaddr'
ok 74 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' <=  'NULL::macaddr'
ok 75 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  =  '''08:00:2b:01:02:03''::macaddr'
ok 76 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  =  '''08:10:2b:01:02:03''::macaddr'
ok 77 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  =  '''08:20:2b:01:02:03''::macaddr'
ok 78 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  =  'NULL::macaddr'
ok 79 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  >= '''08:00:2b:01:02:03''::macaddr'
ok 80 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  >= '''08:10:2b:01:02:03''::macaddr'
ok 81 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  >= '''08:20:2b:01:02:03''::macaddr'
ok 82 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'  >= 'NULL::macaddr'
ok 83 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'   > '''08:00:2b:01:02:03''::macaddr'
ok 84 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'   > '''08:10:2b:01:02:03''::macaddr'
ok 85 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'   > '''08:20:2b:01:02:03''::macaddr'
ok 86 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr'   > 'NULL::macaddr'
ok 87 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' !=  '''08:00:2b:01:02:03''::macaddr'
ok 88 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' !=  '''08:10:2b:01:02:03''::macaddr'
ok 89 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' !=  '''08:20:2b:01:02:03''::macaddr'
ok 90 - check variant vs casted for '''08:10:2b:01:02:03''::macaddr' !=  'NULL::macaddr'
